packages/material-ui/src/Collapse/Collapse.test.js

```diff
@@ -70,279 +53,136 @@ describe('<Collapse />', () => {
     );
   });
 
-  describe('event callbacks', () => {
-    it('should fire event callbacks', () => {
-      const events = ['onEnter', 'onEntering', 'onEntered', 'onExit', 'onExiting', 'onExited'];
-
-      const handlers = events.reduce((result, n) => {
-        result[n] = spy();
-        return result;
-      }, {});
-
-      const wrapper = shallow(<Collapse {...props} {...handlers} />);
-
-      events.forEach(n => {
-        const event = n.charAt(2).toLowerCase() + n.slice(3);
-        wrapper.simulate(event, { style: {} });
-        assert.strictEqual(handlers[n].callCount, 1, `should have called the ${n} handler`);
-      });
-    });
-  });
-
-  describe('prop: timeout', () => {
-    let wrapper;
-    let instance;
-    let element;
-    const enterDuration = 556;
-    const leaveDuration = 446;
-
-    beforeEach(() => {
-      wrapper = shallow(
-        <Collapse
-          {...props}
-          timeout={{
-            enter: enterDuration,
-            exit: leaveDuration,
-          }}
-        />,
-      );
-      instance = wrapper.dive().instance();
-      element = { getBoundingClientRect: () => ({}), style: {} };
-    });
-
-    it('should create proper easeOut animation onEntering', () => {
-      instance.handleEntering(element);
-      assert.strictEqual(element.style.transitionDuration, `${enterDuration}ms`);
-    });
-
-    it('should create proper sharp animation onExiting', () => {
-      instance.handleExiting(element);
-      assert.strictEqual(element.style.transitionDuration, `${leaveDuration}ms`);
-    });
-  });
-
   describe('transition lifecycle', () => {
     let wrapper;
-    let instance;
+    let container;
+    let clock;
+    let nodeEnterHeightStyle;
+    let nodeEnteringHeightStyle;
+    let nodeExitHeightStyle;
+
+    const handleEnter = spy();
+    const handleEnterWrapper = (...args) => {
+      handleEnter(...args);
+      nodeEnterHeightStyle = args[0].style.height;
```

Comment:
I need to add a comment on why this is necessary. (Arguments are passed by reference)